### PR TITLE
Fix context overflow

### PR DESCRIPTION
**Pull request by reciprocated**
Fix of #130, this PR
- limits generation's size to `max_new_tokens`
- truncates prompts to `seq_length - max_new_tokens`
- enables separate logging of prompts and generations

[PPO](https://wandb.ai/sorry/public/reports/Fix-context-overflow-131--VmlldzozMTUyMDI4)
[ILQL](https://wandb.ai/sorry/public/reports/Fix-context-overflow-131--VmlldzozMTUyMDU3)

### CODE REVIEW

File: trlx/model/accelerate_base_model.py [hunk 1]
```diff
@@ -166,23 +167,43 @@ def evaluate(self):
...
+            )
         stats["generate_time"] = time() - generate_time
 
         samples = self.accelerator.gather(torch.vstack(all_samples))
+        prompts_sizes = self.accelerator.gather(torch.hstack(prompts_sizes))
 
         if self.accelerator.is_main_process:
             if self.tokenizer:
-                samples = self.tokenizer.batch_decode(samples, skip_special_tokens=True)
+                str_samples = self.tokenizer.batch_decode(
```
Comment (jon-tow) [inline trlx/model/accelerate_base_model.py:180]: Nice re-naming 👍

**Review by jon-tow — COMMENTED** [review]
Tested locally and looks great! I left a few comments and questions 🙏

File: trlx/model/nn/ilql_models.py [hunk 2]
```diff
@@ -42,8 +42,8 @@ class ILQLConfig(MethodConfig):
     awac_scale: float
     alpha: float
     steps_for_target_q_sync: float
-    betas: Sequence[float]
```
[the remark this answers is not in the document]
Comment (reciprocated) [reply trlx/model/nn/ilql_models.py:45]: previously there were multiple values and for each an independent generate() was called at evaluation time. At some point I started using `betas[0]` as a single beta

**Review by jon-tow — APPROVED** [review]
Looks good to me! Thanks for clearing up my questions.

**Comment by LouisCastricato** [comment]
When are we merging

File: trlx/model/nn/ppo_models.py [hunk 3]
```diff
@@ -162,10 +163,12 @@ def loss(
             old_values - self.cliprange_value,
             old_values + self.cliprange_value,
         )
+        n = mask.sum()
```
Comment (Dahoas) [inline trlx/model/nn/ppo_models.py:166]: I'm not sure but shouldn't n be vector valued where each component is the size of the ith generation? (Before endoftext)

Comment (reciprocated) [reply trlx/model/nn/ppo_models.py:166]: it is used only in reduction to scalars so no

**Review by jon-tow — COMMENTED** [review]
I left one tiny nit if you could take a look!

**Pull request merged**